dbt/macros/crosswalk_year_fill.sql

```diff
@@ -0,0 +1,202 @@
+/*
```

Comment:
The comment and code in this macro is identical to the previous `location.crosswalk_year_fill` definition in [#719](https://github.com/ccao-data/data-architecture/pull/719/files#diff-4c39447a5466b05068a38c9b0cbda6eb4e973ea42af22ddc19ea9b389ed329a1), just with the addition of a macro interface that requires the caller to specify the `schema` string and the `fields` data object.